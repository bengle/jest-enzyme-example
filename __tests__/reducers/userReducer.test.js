import UserReducer from "../../src/reducers/userReducer";

const initState = {
  user: {
    username: "",
    sex: "男",
    age: 20
  },
  list: []
};

describe("user reducer", () => {
  test("init is correct", () => {
    const action = { type: "default" };
    expect(UserReducer(undefined, action)).toEqual(initState);
    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });
  test("test username change reducer", () => {
    const action = {
      type: "CHANGE_USERNAME",
      payload: "abc"
    };
    expect(UserReducer(initState, action).user.username).toBe("abc");
    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });
  test("test sex change reducer", () => {
    const action = {
      type: "CHANGE_SEX",
      payload: true
    };
    expect(UserReducer(initState, action).user.sex).toBe("女");
    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });
  test("test get user list reducer", () => {
    const expectData = [
      {
        username: "tom",
        sex: "男",
        age: 18
      },
      {
        username: "jack",
        sex: "男",
        age: 30
      },
      {
        username: "lucy",
        sex: "女",
        age: 25
      }
    ];
    const action = {
      type: "GET_ALL_USERS",
      payload: expectData
    };
    expect(UserReducer(initState, action).list).toEqual(expectData);
    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });
});
