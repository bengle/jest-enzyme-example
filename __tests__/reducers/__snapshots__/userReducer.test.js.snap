// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user reducer init is correct 1`] = `
Object {
  "list": Array [],
  "user": Object {
    "age": 20,
    "sex": "男",
    "username": "",
  },
}
`;

exports[`user reducer test get user list reducer 1`] = `
Object {
  "list": Array [
    Object {
      "age": 18,
      "sex": "男",
      "username": "tom",
    },
    Object {
      "age": 30,
      "sex": "男",
      "username": "jack",
    },
    Object {
      "age": 25,
      "sex": "女",
      "username": "lucy",
    },
  ],
  "user": Object {
    "age": 20,
    "sex": "男",
    "username": "",
  },
}
`;

exports[`user reducer test sex change reducer 1`] = `
Object {
  "list": Array [],
  "user": Object {
    "age": 20,
    "sex": "女",
    "username": "",
  },
}
`;

exports[`user reducer test username change reducer 1`] = `
Object {
  "list": Array [],
  "user": Object {
    "age": 20,
    "sex": "男",
    "username": "abc",
  },
}
`;
